src/DateHelpers.elm

```diff
@@ -0,0 +1,361 @@
+            toMilliseconds Jan 1 (Time.toYear timeZone posix) - 1
+
+        dayDifferenceFromUTC =
+            Time.toDay utc posix - Time.toDay timeZone posix
+
+        -- if we cross a month boundary, we want to flip the offset back since sutraction caused the polarity to flip
+        offset =
+            if Basics.abs dayDifferenceFromUTC > 1 then
+                -1 * clamp -1 1 dayDifferenceFromUTC
+
+            else
+                dayDifferenceFromUTC
+
+        millis =
+            Time.posixToMillis posix
+
+        diff =
+            millis - oneMillisecondBeforeMidnight
+    in
+    ceiling (toFloat diff / 86400000) - offset
+
+
+dayOfYearToPosix : Int -> Int -> Time.Posix
+dayOfYearToPosix year doy =
+    let
+        firstOfYear =
+            toMilliseconds Jan 1 year
+
+        millis =
+            firstOfYear + ((doy - 1) * 86400000)
+    in
+    Time.millisToPosix millis
+
+
+
+-- Date calculation taken from https://cs.uwaterloo.ca/~alopez-o/math-faq/node73.html
+
+
+getDayOfWeek : Month -> Int -> Int -> Int
+getDayOfWeek month day year =
+    let
+        century =
+            firstNDigits 2 year
+
+        monthVal =
+            monthToDayOfWeekCalculationValue month
+
+        --  last two digits of the year. Jan and Feb are treated as the previous year
+        yearVal =
+            case month of
+                Jan ->
+                    modBy 100 (year - 1)
+
+                Feb ->
+                    modBy 100 (year - 1)
+
+                _ ->
+                    modBy 100 year
+    in
+    (day + floor (2.6 * toFloat monthVal - 0.2) - 2 * century + yearVal + floor (toFloat yearVal / 4) + floor (toFloat century / 4)) |> modBy 7
+
+
+firstNDigits : Int -> Int -> Int
+firstNDigits n num =
+    num // (10 ^ (truncate (logBase 10 (toFloat num)) - n + 1))
+
+
+monthToStandardValue : Month -> Int
+monthToStandardValue month =
+    case month of
+        Jan ->
+            1
+
+        Feb ->
+            2
+
+        Mar ->
+            3
+
+        Apr ->
+            4
+
+        May ->
+            5
+
+        Jun ->
+            6
+
+        Jul ->
+            7
+
+        Aug ->
+            8
+
+        Sep ->
+            9
+
+        Oct ->
+            10
+
+        Nov ->
+            11
+
+        Dec ->
+            12
+
+
+
+-- Treat jan and feb as 11 and 12 of the previous year to make the calculation work
+
+
+monthToDayOfWeekCalculationValue : Month -> Int
+monthToDayOfWeekCalculationValue month =
+    case month of
+        Jan ->
+            11
+
+        Feb ->
+            12
+
+        Mar ->
+            1
+
+        Apr ->
+            2
+
+        May ->
+            3
+
+        Jun ->
+            4
+
+        Jul ->
+            5
+
+        Aug ->
+            6
+
+        Sep ->
+            7
+
+        Oct ->
+            8
+
+        Nov ->
+            9
+
+        Dec ->
+            10
+
+
+monthToString : Month -> String
+monthToString month =
```

Comment:
You can simplify `monthToStandardValue << String.fromInt` reusing the prev function

Reply:
Wow, can't believe I missed that! Thanks!